src/main/java/com/minecolonies/core/colony/buildings/workerbuildings/BuildingMiner.java

```diff
@@ -159,26 +166,25 @@ public CompoundTag serializeNBT()
      */
     public int getDepthLimit(final Level level)
     {
-        if (this.getBuildingLevel() == 1)
-        {
-            return normalizeMaxDepth(MAX_DEPTH_LEVEL_1, level);
-        }
-        else if (this.getBuildingLevel() == 2)
-        {
-            return normalizeMaxDepth(MAX_DEPTH_LEVEL_2, level);
-        }
-        else if (this.getBuildingLevel() == 3)
-        {
-            return normalizeMaxDepth(MAX_DEPTH_LEVEL_3, level);
-        }
-        else if (this.getBuildingLevel() == 4)
-        {
-            return normalizeMaxDepth(MAX_DEPTH_MAX, level);
-        }
-        else
+        int buildingY = this.getLadderLocation().getY() - 5;
+
+        int buildingLevels = getBuildingLevel();
+        int yLevel = 0;
+        for (final Integer miningLevel : MINING_LEVELS)
         {
-            return normalizeMaxDepth(-100, level);
+            if (miningLevel < buildingY)
+            {
+                yLevel = miningLevel;
+                buildingLevels--;
+            }
+
+            if (buildingLevels == 0)
```

Comment:
This will loop badly if `getBuildingLevel() == 0` (since `buildingLevels` becomes negative), though the loop will eventually terminate since it's bound by the array.

Reply:
AI should never query this at level 0, but maybe some early return at <= 0 could be nice.

Reply:
well it loops a whopping 4 entries xD so unless you need a specific return on 0 it probably doesnt matter